src/Symfony/Component/Console/Tests/Helper/TableTest.php

```diff
@@ -252,6 +253,214 @@ public function testRenderProvider()
+
+TABLE
+            ),
+            'Cell with rowspan and colspan' => array(
+                array('ISBN', 'Title', 'Author'),
+                array(
+                    array(
+                        new TableCell('9971-5-0210-0', array('rowspan' => 2, 'colspan' => 2)),
+                        'Dante Alighieri',
+                    ),
+                    array('Charles Dickens'),
+                    new TableSeparator(),
+                    array(
+                        'Dante Alighieri',
+                        new TableCell('9971-5-0210-0', array('rowspan' => 3, 'colspan' => 2)),
+                    ),
+                    array('J. R. R. Tolkien'),
+                    array('J. R. R'),
+                ),
+                'default',
+<<<TABLE
++------------------+--------+-----------------+
+| ISBN             | Title  | Author          |
++------------------+--------+-----------------+
+| 9971-5-0210-0             | Dante Alighieri |
+|                           | Charles Dickens |
++------------------+--------+-----------------+
+| Dante Alighieri  | 9971-5-0210-0            |
+| J. R. R. Tolkien |                          |
+| J. R. R          |                          |
++------------------+--------+-----------------+
+
+TABLE
+            ),
+            'Cell with rowspan and colspan contains new line break' => array(
+                array('ISBN', 'Title', 'Author'),
+                array(
+                    array(
+                        new TableCell("9971\n-5-\n021\n0-0", array('rowspan' => 2, 'colspan' => 2)),
+                        'Dante Alighieri',
+                    ),
+                    array('Charles Dickens'),
+                    new TableSeparator(),
+                    array(
+                        'Dante Alighieri',
+                        new TableCell("9971\n-5-\n021\n0-0", array('rowspan' => 2, 'colspan' => 2)),
+                    ),
+                    array('Charles Dickens'),
+                    new TableSeparator(),
+                    array(
+                        new TableCell("9971\n-5-\n021\n0-0", array('rowspan' => 2, 'colspan' => 2)),
+                        new TableCell("Dante \nAlighieri", array('rowspan' => 2, 'colspan' => 1)),
+                    ),
+                ),
+                'default',
+<<<TABLE
++-----------------+-------+-----------------+
+| ISBN            | Title | Author          |
++-----------------+-------+-----------------+
+| 9971                    | Dante Alighieri |
+| -5-                     | Charles Dickens |
+| 021                     |                 |
+| 0-0                     |                 |
++-----------------+-------+-----------------+
+| Dante Alighieri | 9971                    |
+| Charles Dickens | -5-                     |
+|                 | 021                     |
+|                 | 0-0                     |
++-----------------+-------+-----------------+
```

Comment:
I would expect:

```
+-----------------+-------+-----------------+
| Dante Alighieri | 9971                    |
| Charles Dickens | -5-0210-0               |
+-----------------+-------+-----------------+
```

You get the point. The text should take the maximum space available in the merged cells. Does it make sense?